src/common/marlin_server.hpp

```diff
@@ -13,7 +13,13 @@ void fsm_create(ClinetFSM type, uint8_t data);
 //notify all clients to destroy finit statemachine, must match fsm_destroy_t signature
 void fsm_destroy(ClinetFSM type);
 //notify all clients to change state of finit statemachine, must match fsm_change_t signature
-void fsm_change(ClinetFSM type, uint8_t phase, uint8_t progress_tot, uint8_t progress);
+//can be called inside while, notification is send only when is different from previous one
+void _fsm_change(ClinetFSM type, uint8_t phase, uint8_t progress_tot, uint8_t progress);
+
+template <class T>
+void fsm_change(ClinetFSM type, T phase, uint8_t progress_tot, uint8_t progress) {
+    _fsm_change(type, GetPhaseIndex(phase), progress_tot, progress);
+}
```

Comment:
How many phases do we have in FW? Is GetPhaseIndex marked as constexpr?

Reply:
Load dialog has 17 phases
other dialogs are not refactored yet

Reply:
yes, it is